Update RFC 7589 (#11)
netconf-wg/netconf-over-tls13: draft-ietf-netconf-over-tls13.md

--- abstract

RFC 7589 defines how to protect NETCONF messages with TLS 1.2. This
document describes how to protect NETCONF messages with TLS 1.3. This
document also updates RFC 7589 to address support requirements for

> TLS 1.3.

Author: ```suggestion
the mandatory to implement TLS 1.2 cipher suites and TLS 1.3.
```

> TLS 1.3.

Author: ```suggestion
the mandatory to implement TLS 1.2 cipher suite and TLS 1.3.
```

> document updates RFC 8325 to address TLS 1.2 and TLS 1.3 cipher
> suites, support requirements for TLS 1.2 and TLS 1.3, and the
> use of TLS 1.3's early data, which is also known as 0-RTT data.

Author: ```suggestion
document updates RFC 8325 to address support requirements for TLS 1.2
and TLS 1.3 and the use of TLS 1.3's early data.
```

> the mandatory to implement TLS 1.2 cipher suite and TLS 1.3.

Author: ```suggestion
RFC 7589 defines how to protect NETCONF messages with TLS 1.2. This
document updates RFC 8325 to address TLS 1.2 and TLS 1.3 cipher
suites, support requirements for TLS 1.2 and TLS 1.3, and the
use of TLS 1.3's early data, which is also known as 0-RTT data.
```

--- middle

# Introduction

{{!RFC7589}} defines how to protect NETCONF messages {{!RFC6241}} with

> TLS 1.2 {{!RFC5246}}. This document describes defines how to protect

Author: ```suggestion
TLS 1.2 {{!RFC5246}}. This document defines how to protect
```

NETCONF messages with TLS 1.3 {{!I-D.ietf-tls-rfc8446bis}}. This
document also updates {{RFC7589}} to address support requirements for

> TLS 1.3 {{!I-D.ietf-tls-rfc8446bis}}.

Author: ```suggestion
the mandatory to implement TLS 1.2 cipher suite and TLS 1.3 {{!I-D.ietf-tls-rfc8446bis}}.
```

# Cipher Suites

Implementations MUST support TLS 1.2 {{RFC5246}} and are REQUIRED to

> support the TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256 cipher suite {{RFC9325}}.

Author: ```suggestion
support the TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256 cipher suite {{!RFC9325}}.
```

# Cipher Suites

Implementations MUST support TLS 1.2 {{RFC5246}} and are REQUIRED to

> support the mandatory-to-implement cipher suite.

Author: ```suggestion
support the TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256 cipher suite {{RFC9325}}.
```

Author: Leaving the MIT pinned to TLS 1.2's MTI, which is TLS_RSA_WITH_AES_128_CBC_SHA, would never get through the IESG. Should be pinned to RFC 9325.

> Implementations MAY implement additional TLS cipher suites that provide

Author: ```suggestion
Implementations MAY implement additional TLS 1.2 cipher suites that provide
```

mutual authentication {{RFC5246}} and confidentiality as required by
NETCONF {{RFC6241}}.

> Implementations SHOULD support TLS 1.3 {{I-D.ietf-tls-rfc8446bis}} and,
> if implemented, MUST prefer to negotiate TLS 1.3 over earlier versions
> of TLS.

Author: ```suggestion
Implementations that support more than one version of TLS MUST
prefer later versions of TLS over earlier versions of TLS.
```

Author: Russ suggested this version of the text to better future proof it.

Author: Decided to leave it alone. When there's a new TLS version there will no doubt be a new cipher suite so this section will need to change regardless.
